sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/StatsbeatTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Monitor.OpenTelemetry.Exporter.Internals;
+using Azure.Monitor.OpenTelemetry.Exporter.Internals.ConnectionString;
+using Xunit;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Tests
+{
+    public class StatsbeatTests
+    {
+        [Theory]
+        [InlineData("francecentral")]
+        [InlineData("francesouth")]
+        [InlineData("northeurope")]
+        [InlineData("norwayeast")]
+        [InlineData("norwaywest")]
+        [InlineData("swedencentral")]
+        [InlineData("switzerlandnorth")]
+        [InlineData("switzerlandwest")]
+        [InlineData("uksouth")]
+        [InlineData("ukwest")]
+        [InlineData("westeurope")]
+        public void StatsbeatConnectionStringIsSetBasedOnCustomersConnectionStringEndpointInEU(string euEndpoint)
+        {
+            var customer_ConnectionString = $"InstrumentationKey=1aa11111-bbbb-1ccc-8ddd-eeeeffff3333;IngestionEndpoint=https://{euEndpoint}.in.applicationinsights.azure.com/";
+            var parsedConectionString = ConnectionStringParser.GetValues(customer_ConnectionString);
+
+            // Set ConnectionString to null as it is static and may be impacted by other tests.
+            Statsbeat.s_statsBeat_ConnectionString = null;
+            Statsbeat.SetStatsbeatConnectionString(parsedConectionString.IngestionEndpoint);
+
+            Assert.Equal(Statsbeat.StatsBeat_ConnectionString_EU, Statsbeat.s_statsBeat_ConnectionString);
+        }
+
+        [Theory]
+        [InlineData("australiacentral")]
+        [InlineData("australiacentral2")]
+        [InlineData("australiaeast")]
+        [InlineData("australiasoutheast")]
+        [InlineData("brazilsouth")]
+        [InlineData("brazilsoutheast")]
+        [InlineData("canadacentral")]
+        [InlineData("canadaeast")]
+        [InlineData("centralindia")]
+        [InlineData("centralus")]
+        [InlineData("chinaeast2")]
+        [InlineData("chinaeast3")]
+        [InlineData("chinanorth3")]
+        [InlineData("eastasia")]
+        [InlineData("eastus")]
+        [InlineData("eastus2")]
+        [InlineData("japaneast")]
+        [InlineData("japanwest")]
+        [InlineData("jioindiacentral")]
+        [InlineData("jioindiawest")]
+        [InlineData("koreacentral")]
+        [InlineData("koreasouth")]
+        [InlineData("northcentralus")]
+        [InlineData("qatarcentral")]
+        [InlineData("southafricanorth")]
+        [InlineData("southcentralus")]
+        [InlineData("southeastasia")]
+        [InlineData("southindia")]
+        [InlineData("uaecentral")]
+        [InlineData("uaenorth")]
+        [InlineData("westus")]
+        [InlineData("westus2")]
+        [InlineData("westus3")]
```

Comment:
I think InlineData can accept an array of values. May be able to provide the constant from the other class.

Reply:
We don't want the array here right? It needs to be tested one by one. I could have a loop inside the test but that will look odd. Let me know if I am missing something. I will follow up on that with different PR.

Reply:
Something like this one - https://github.com/Azure/azure-sdk-for-net/blob/045f589c736a6c76af682c3ac01f684b8da93a5e/sdk/core/Azure.Core/tests/RedirectPolicyTests.cs#L21

Reply:
e.g. https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/#loading-data-from-a-property-on-the-test-class

Reply:
Ran in to some issue, will do this as a follow up.